applications/GeoMechanicsApplication/python_scripts/geomechanics_solver.py

```diff
@@ -0,0 +1,483 @@
+    #         self._set_and_fill_buffer()
+
+    #     # This will be removed once the Model is fully supported! => It wont e necessary anymore
+    #     if not self.model.HasModelPart(self.main_model_part.Name):
+    #         self.model.AddModelPart(self.main_model_part)
+
+    #     KratosMultiphysics.Logger.PrintInfo("::[GeoMechanicalSolver]::", "ModelPart prepared for Solver.")
+
+    def Initialize(self):
+        """Perform initialization after adding nodal variables and dofs to the main model part. """
+        KratosMultiphysics.Logger.PrintInfo("::[GeoMechanicalSolver]:: ", "Initializing ...")
+        # The mechanical solution strategy is created here if it does not already exist.
+        if self.settings["clear_storage"].GetBool():
+            self.Clear()
+        mechanical_solution_strategy = self.get_mechanical_solution_strategy()
+        mechanical_solution_strategy.SetEchoLevel(self.settings["echo_level"].GetInt())
+        if not self.is_restarted():
+            mechanical_solution_strategy.Initialize()
+        else:
+            # SetInitializePerformedFlag is not a member of SolvingStrategy but
+            # is used by ResidualBasedNewtonRaphsonStrategy.
+            try:
+                mechanical_solution_strategy.SetInitializePerformedFlag(True)
+            except AttributeError:
+                pass
+        self.Check()
+        KratosMultiphysics.Logger.PrintInfo("::[GeoMechanicalSolver]:: ", "Finished initialization.")
+
+    # def Solve(self):
+    #     KratosMultiphysics.Logger.PrintWarning("::[GeoMechanicalSolver]:: ", "Solve(self)")
+    #     if self.settings["clear_storage"].GetBool():
+    #         self.Clear()
+    #     mechanical_solution_strategy = self.get_mechanical_solution_strategy()
+    #     mechanical_solution_strategy.Solve()
+
+    def InitializeSolutionStep(self):
+        self.get_mechanical_solution_strategy().InitializeSolutionStep()
+
+    def Predict(self):
+        self.get_mechanical_solution_strategy().Predict()
+
+    def SolveSolutionStep(self):
+        is_converged = self.get_mechanical_solution_strategy().SolveSolutionStep()
+        return is_converged
+
+    def FinalizeSolutionStep(self):
+        self.get_mechanical_solution_strategy().FinalizeSolutionStep()
+
+    def AdvanceInTime(self, current_time):
+        dt = self.ComputeDeltaTime()
+        new_time = current_time + dt
+        self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] += 1
+        self.main_model_part.CloneTimeStep(new_time)
+
+        return new_time
+
+    def ComputeDeltaTime(self):
+        return self.settings["time_stepping"]["time_step"].GetDouble()
+
+    def GetComputingModelPart(self):
+        if not self.main_model_part.HasSubModelPart(self.settings["computing_model_part_name"].GetString()):
+            raise Exception("The ComputingModelPart was not created yet!")
+        return self.main_model_part.GetSubModelPart(self.settings["computing_model_part_name"].GetString())
```

Comment:
this was removed in Structural
it was a huge pain, I strongly recommend doing the same